client/pages/post.js

```diff
@@ -31,14 +31,13 @@ class PostDetailPage extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      subreddit: this.props.subreddit,
-      user: this.props.user,
+      subreddit: this.props.post.subreddit,
+      user: this.props.post.user,
```

Comment:
the subreddit and user (the logged in one) here flow through to the sidebar components, to let them do their thing.  do you also need a user (the post creator) and subreddit, but the whole object from the post?